back/src/forms/validation.ts

```diff
@@ -1347,10 +1347,6 @@ const withNextDestination = (required: boolean) =>
       nextDestinationCompanyExtraEuropeanId: yup.string().nullable(),
       nextDestinationNotificationNumber: yup
         .string()
```

Comment:
```suggestion
        .string()
        .when("wasteDetailsCode", {
          is: wasteDetailsCode =>
            isDangerous(wasteDetailsCode),
          then: schema =>
            schema.matches(/^[a-zA-Z]{2}[0-9]{10}$|^$/,
          "Destination ultérieure : Le numéro d'identification ou de document doit être au format au format PP AAAA DDDRRR, composé de 2 lettres (code pays) puis 4 chiffres (année) puis DDD (départment de départ, par exemple 075) puis RRR (numéro d'ordre)"),
          otherwise: schema => schema.matches(/^[a-zA-Z]{2}[0-9]{4}$|^$/,
          "Destination ultérieure : Le numéro d'identification ou de document doit être au format au format A7E AAAA DDDRRR, composé de A7E puis 4 chiffres (année) puis DDD (départment de départ, par exemple 075) puis RRR (numéro d'ordre)"),
        })
```